### PR TITLE
Upgrading to Realm Core 0.97.0

### PR DESCRIPTION
**Pull request by kneth**
@realm/java

### CODE REVIEW

**Comment by beeender** [comment]
Not sure if https://github.com/realm/realm-object-store/pull/52/files#r55271290 this change has impact on us as well. in `completedAsyncQueriesUpdate`

**Comment by kneth** [comment]
@beeender The error in the failing test seems to indicate it :-)

**Comment by cmelchior** [comment]
Should this PR also modify our `deleteRealm` method? since (all?) the auxillary files are now inlined?

**Comment by kneth** [comment]
@cmelchior We can remove the aux. files from the list but then removing old database will leave these files untouched. Adding a note to the change log that you must manually remove aux. files is an option.

**Comment by cmelchior** [comment]
I guess leaving it in for now isn't a big issue.

File: realm/realm-library/src/main/java/io/realm/internal/SharedGroupManager.java [hunk 1]
```diff
@@ -169,10 +169,12 @@ public static boolean compact(RealmConfiguration configuration) {
         try {
             sharedGroup = new SharedGroup(
                     configuration.getPath(),
-                    SharedGroup.EXPLICIT_TRANSACTION,
+                    SharedGroup.IMPLICIT_TRANSACTION,
```
Comment (nhachicha) [inline realm/realm-library/src/main/java/io/realm/internal/SharedGroupManager.java]: Why?

File: realm/realm-library/src/main/java/io/realm/internal/SharedGroupManager.java [hunk 2]
```diff
@@ -169,10 +169,12 @@ public static boolean compact(RealmConfiguration configuration) {
...
                     SharedGroup.Durability.FULL,
                     configuration.getEncryptionKey());
             result = sharedGroup.compact();
+        } catch (Exception e) {
+            return false;
```
Comment (nhachicha) [inline realm/realm-library/src/main/java/io/realm/internal/SharedGroupManager.java]: Shouldn't we at least log the error _RealmLog_

File: realm/realm-jni/src/util.hpp [hunk 3]
```diff
@@ -101,7 +101,7 @@ std::string num_to_string(T pNumber)
 #define G(x)    reinterpret_cast<realm::Group*>(x)
 #define ROW(x)  reinterpret_cast<realm::Row*>(x)
 #define SG(ptr) reinterpret_cast<realm::SharedGroup*>(ptr)
-#define CH(ptr) reinterpret_cast<realm::ClientHistory*>(ptr)
+#define CH(ptr) reinterpret_cast<realm::Replication*>(ptr)
```
Comment (nhachicha) [inline realm/realm-jni/src/util.hpp]: I don't think this macro is used anymore. remove it?

**Comment by nhachicha** [comment]
LGTM besides small comments

**Comment by beeender** [comment]
The `TableView`'s version could be `std::numeric_limits<uint64_t>::max()` for the default of the source `LinkView` gets deleted. See https://github.com/realm/realm-core/pull/1510 . After the max `uint64_t` convert to jlong, it give `-1` which is accidentally the same with our `RealmResults.TABLE_VIEW_VERSION_NONE`. I checked the code, and it seems to be no problem with current implementation. However, @nhachicha would you please help to check again? This can be hard to debug if there is a problem in the future.

**Comment by cmelchior** [comment]
I think that `std::numeric_limits<uint64_t>::max()` have always been the value used internally by Core, but now it just got more visible.

File: realm/realm-library/src/main/java/io/realm/internal/SharedGroupManager.java (hunk 1 above)
Comment (kneth) [inline realm/realm-library/src/main/java/io/realm/internal/SharedGroupManager.java]: We get a `LogicError` exception when you try to open the Realm with explicit transaction if it is already opened with implicit transaction. I haven't talked with the core team so I don't know if they are aware of it.

File: realm/realm-jni/src/util.hpp (hunk 3 above)
Comment (kneth) [inline realm/realm-jni/src/util.hpp]: It is used three places in `io_realm_internal_SharedGroup.cpp`.

**Comment by cmelchior** [comment]
:+1:

**Comment by nhachicha** [comment]
:+1:

**Comment by beeender** [comment]
:+1:

**Comment by stk1m1** [comment]
:+1:

**Pull request merged**